crates/autopilot/src/run_loop.rs

```diff
@@ -187,26 +199,41 @@ impl RunLoop {
                 },
                 solutions: solutions
                     .iter()
-                    .map(|(index, response)| SolverSettlement {
-                        solver_address: response.submission_address,
-                        score: Some(Score::Solver(response.score)),
-                        ranking: Some(solutions.len() - index),
-                        orders: response
-                            .orders
-                            .iter()
-                            .map(|o| Order {
-                                id: *o,
-                                // TODO: revisit once colocation is enabled (remove not populated
-                                // fields) Not all fields can be
-                                // populated in the colocated world
-                                ..Default::default()
-                            })
-                            .collect(),
-                        call_data: response.calldata.internalized.clone(),
-                        uninternalized_call_data: Some(response.calldata.uninternalized.clone()),
-                        // TODO: revisit once colocation is enabled (remove not populated fields)
-                        // Not all fields can be populated in the colocated world
-                        ..Default::default()
+                    .map(|(index, response)| {
+                        let (orders, call_data, uninternalized_call_data) =
```

Comment:
Might be slightly nicer to read if it would look like this:
```
let mut settlement = SolverSettlement{..};
if is_winner {
    settlement.orders = ...;
    ...
}
return settlement;
```

Reply:
Indeed, refactored.